Solutions/Service Definitions/ManagementService.yaml

```diff
@@ -299,6 +463,84 @@ components:
               type: string
               enum: ['notTracked', 'unread', 'read']
 
+    SmsTemplate:
+      description: The sms object.
+      properties:
+        body:
+          description: The inner templated body for the sms object.
+          nullable: false
+          type: string
+
+    EmailTemplate:
+      description: The email object.
+      properties:
+        body:
+          description: The inner templated body for the email object.
+          nullable: false
+          type: string
+        subject:
+          description: The subject of the email object.
+          nullable: false
+          type: string
+        important:
+          description: Mark is the email is important.
+          nullable: true
+          type: string
+          
+    WebPushTemplate:
+      description: The webpush object.
+      properties:
+        body:
+          description: The inner templated body for the email object.
+          nullable: false
+          type: string
+        title:
+          description: The title of the email object.
+          nullable: false
+          type: string
+        image:
+          description: The Base64 image of webpush notification.
+          nullable: true
+          type: string
+
+    NotificationTemplate:
+      description: The template wrapper which will stored per notification type.
+      required:
+        - notificationType
+      properties:
+        notificationType:
+          description: The notification type for the template.
+          nullable: false
+          type: string
+        smsTemplate:
```

Comment:
I don't really like hardcoding these template types in here. Is this intended to be a union type (i.e. oneOf) or are you expecting to store several of these?

If one of, then a property for a discriminator content type string and a property for the payload (whose schema is determined by the content type) would be better.

If several, then a more extensible approach would be an object whose keys are the content type of the template, and whose values are the templates themselves (again, using the the content type of the template).

You could still include the schema for the payloads in the your OpenAPI schema, and require a `const` value for their `contentType` property of the relevant content type so you clearly (and machine-readably) document the intent.

Reply:
Yeah. We will be looking at this tommorow first thing in the morning.

Reply:
So now we dont have hardcoded template. We have used anyOf now in the YAML. We did try to use oneOf but for some reason we could not get the request object correctly serialised into the correct model type. 

We used exactly the discriminator with the contentType to distinguish between different templates.